src/main/scala/org/allenai/ike/IkeBatchSearch.scala

```diff
@@ -0,0 +1,338 @@
+      tempDir.deleteOnExit()
+      tempDir
+    }
+
+    // Split the patterns and key them by table name. We also validate the pattern lines here.
+    val patternLines = sparkContext.textFile(batchSearchOptions.patternFile).collect
+    val patternsByTable = patternLines.flatMap { patternLine =>
+      // Process the pattern and run the query.
+      splitPattern(patternLine) match {
+        case Some(v) => Some(v)
+        case _ =>
+          logger.error("Invalid format in pattern file, line: " + patternLine)
+          None
+      }
+    }.groupBy(_._2).map {
+      case (tableName, splitLines) =>
+        if (splitLines.length == 1) {
+          (tableName, splitLines.head._1)
+        } else {
+          logger.error("Multiple patterns mapping to the same output table: " + tableName)
+          sys.exit(1)
+        }
+    }
+
+    // Find all existing index directories. If there are none, we'll need to build an index.
+    // TODO: Resuming a partial index would be a nice feature.
+    val indexDirectories = indexParentDir.listFiles().filter(_.isDirectory)
+    val indexPaths = if (indexDirectories.nonEmpty) {
+      indexDirectories.map(_.getAbsolutePath)
+    } else {
+      // Load the text source and partition it according to ideal index size. The maximum sentences
+      // per partition were determined through several experimental runs. As indices get larger,
+      // indexing slows down. Since partitioning in our scenario is arbitrary and relevance doesn't
+      // matter, we should optimize strictly for speed.
+      val textLines = sparkContext.textFile(batchSearchOptions.textSource)
+      val sentences = textLines.repartition {
+        val maxSentencesPerPartition = 256 * 1024
+        val minPartitions = Runtime.getRuntime.availableProcessors
+        val idealPartitions = textLines.count.toDouble / maxSentencesPerPartition
+        ((idealPartitions / minPartitions).ceil * minPartitions).toInt
+      }
+
+      // Create an index for each partition.
+      val indexParentPath = indexParentDir.getAbsolutePath
+      sentences.mapPartitionsWithIndex {
+        case (partIndex, partition) =>
+          val indexDir = Files.createDirectory(Paths.get(indexParentPath + "/" + partIndex)).toFile
+          val indexer = new Indexer(indexDir, true, classOf[AnnotationIndexer])
+          partition.foreach { sentence =>
+            val idText = IdText(batchSearchOptions.textSource, sentence)
+            val processedText = CreateIndex.process(idText, oneSentencePerDoc = true)
+            if (processedText.nonEmpty) {
+              CreateIndex.addTo(indexer)(processedText.head)
+            }
+          }
+          indexer.close()
+
+          Seq(indexDir.getAbsolutePath).toIterator
+      }.collect
+    }
+
+    // For each index and pattern, search and dump the serialized results.
+    val patternResults: Seq[(String, String)] = indexPaths.toSeq.par.flatMap { indexPath =>
```

Comment:
This portion can't be easily Sparkified due to the large number of things that are passed in (tables, etc).  If we rebuilt from the ground up it would be a possibility.